### PR TITLE
Issue #774: Add WIP megatile

### PR DESCRIPTION
**Pull request by mcomella**
Ignore the other bugs reviews. I need to handle telemetry fires so I figured I should land this so others can work on it, or code on top of it, while I work on the fire. It's not perfect:
- The spacing is mostly correct but it seems that there's too much vertical margin in the megatile (are images scaling correctly?), which might cause the cut-off on the bottom
- Round corners of thumbnails
- Focused state of megatile
- Colors of the rest of the overlay (#818).

screenshot_1525124788

### CODE REVIEW

File: app/src/test/java/org/mozilla/focus/home/pocket/PocketEndpointTest.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+private const val KEY_INNER = "list"
+
+@RunWith(RobolectricTestRunner::class)
+class PocketEndpointTest {
+
+    @Test
+    fun `convert Videos JSON successfully to internal objects`() {
+        val expectedSubset = listOf(
+                PocketVideo(
```
Comment (bkmunar) [inline app/src/test/java/org/mozilla/focus/home/pocket/PocketEndpointTest.kt]: Do we need to change this list every time the pocket video recommendations change? Or are we keeping these recommendations constant? I guess my question is are these hard-set recommendations? haha

Comment (mcomella) [reply app/src/test/java/org/mozilla/focus/home/pocket/PocketEndpointTest.kt]: @bkmunar What do you think? What portion of the code is this test testing and how would that relate to the test lists that we provide?

File: app/src/main/java/org/mozilla/focus/home/pocket/Pocket.kt [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+     * different lists.
+     *
+     * This method assumes [startBackgroundUpdates] has been called before being accessed.
+     */
+    @AnyThread // videosCache is synchronized.
+    fun getRecommendedVideos() = videosCache
+    private fun getRecommendedVideosNoCacheAsync() = async { PocketEndpoint.getRecommendedVideos() }
+
+    private fun startBackgroundUpdatesInner() = launch {
+        while (true) {
```
Comment (bkmunar) [inline app/src/main/java/org/mozilla/focus/home/pocket/Pocket.kt]: is this the convention for doing these kinds of background data pulling in apps?

File: app/src/main/java/org/mozilla/focus/home/pocket/PocketVideoFragment.kt [hunk 3]
```diff
@@ -112,8 +117,25 @@ private class PocketVideoAdapter(
         updateForFocusState(holder, holder.itemView.isFocused)
 
         titleView.text = item.title
-        subdomainView.text = "youtube" // TODO: get from Video.
         videoThumbnailView.setBackgroundColor(Color.parseColor("#ee0000")) // TODO: load async.
+
+        val itemURI = try {
+            URI(item.dedupeURL)
+        } catch (e: Exception) { // Apparently Kotlin doesn't have multi-catch.
```
Comment (bkmunar) [inline app/src/main/java/org/mozilla/focus/home/pocket/PocketVideoFragment.kt]: that's so weird, come on kotlin

File: app/src/main/java/org/mozilla/focus/home/pocket/Pocket.kt (hunk 2 above)
Comment (mcomella) [reply app/src/main/java/org/mozilla/focus/home/pocket/Pocket.kt]: That's a good question. I don't know that there is a convention – is there a convention you're aware of, or that you're aware of in iOS?

Oftentimes the data pulling is on-demand but I figured this was a better user experience for what we're trying to achieve, given how frequently the overlay can be shown.

File: app/src/main/java/org/mozilla/focus/home/pocket/Pocket.kt [hunk 4]
```diff
@@ -120,3 +123,10 @@ data class PocketVideo(
         }
     }
 }
+
+fun ImageView.setImageDrawableAsPocketWordmark() {
+    // We want to set SVGs in code because they can produce artifacts otherwise.
```
Comment (bkmunar) [inline app/src/main/java/org/mozilla/focus/home/pocket/Pocket.kt]: what does this mean, what are artifacts?

Comment (mcomella) [reply app/src/main/java/org/mozilla/focus/home/pocket/Pocket.kt]: An unintended visual anomaly generally caused by various implementation details. One example would be when you scale up an image and you can "see the pixels".

In this case, setting SVG in XML can distort it in arbitrary ways (like the ninja star substituting for the settings gear on the homescreen) so I've been blanket setting all SVGs in code.

See https://en.wikipedia.org/wiki/Visual_artifact for more.

File: app/src/main/java/org/mozilla/focus/home/pocket/PocketVideoMegaTile.kt [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import org.mozilla.focus.R
+import org.mozilla.focus.utils.PicassoWrapper
+import kotlin.properties.Delegates
+
+/** A view that contains the Pocket logo and several thumbnails from Pocket videos. */
+class PocketVideoMegaTile(
+        context: Context, attrs: AttributeSet
+) : LinearLayout(context, attrs) {
+
+    var pocketVideos by Delegates.observable<List<PocketVideo>?>(null) { _, _, newVideos ->
```
Comment (bkmunar) [inline app/src/main/java/org/mozilla/focus/home/pocket/PocketVideoMegaTile.kt]: tehe i know this now

File: app/src/main/res/layout/browser_overlay.xml [hunk 6]
```diff
@@ -16,54 +16,7 @@
...
-            android:id="@+id/navButtonSettings"
-            style="@style/NavigationButton"
-            android:nextFocusLeft="@id/turboButton"
-            android:contentDescription="@string/menu_settings"/>
-
-        <ImageView
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
-            android:src="@drawable/ic_firefox_and_workmark" />
+        <include layout="@layout/browser_overlay_top_nav"/>
```
Comment (bkmunar) [inline app/src/main/res/layout/browser_overlay.xml]: this is cool. i like this

File: app/src/main/res/layout/browser_overlay.xml [hunk 7]
```diff
@@ -8,13 +8,19 @@
...
+    >
 
     <LinearLayout
         android:layout_width="match_parent"
-        android:layout_height="0dp"
-        android:layout_weight="1"
-        android:orientation="horizontal">
+        android:layout_height="wrap_content"
+        android:layout_marginStart="57dp"
+        android:layout_marginEnd="57dp"
```
Comment (bkmunar) [inline app/src/main/res/layout/browser_overlay.xml]: do we care about putting these into constants?

File: app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt [hunk 8]
```diff
@@ -93,12 +93,12 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
...
         LayoutInflater.from(context)
                 .inflate(R.layout.browser_overlay, this, true)
-        listOf(navButtonBack, navButtonForward, navButtonReload, navButtonSettings,
-                turboButton, pinButton, pocketVideoMegaTileView)
-                .forEach {
-                    it.setOnClickListener(this)
-                }
-        topNavContainer.forEachChild { it.nextFocusDownId = navUrlInput.id }
+
+        topNavContainer.forEachChild {
```
Comment (bkmunar) [inline app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt]: jk now i know what it's for

File: app/src/main/res/layout/browser_overlay.xml (hunk 7 above)
Comment (mcomella) [reply app/src/main/res/layout/browser_overlay.xml]: No – jumping between multiple files to read constant values is a huge pain (and it pollutes the xml files they're defined in). It's useful to extract values when:
- They're used many, many times
- They change across device configurations
- They're needed in code

**Comment by mcomella** [comment]
Tests are green, need to rebase on latest master but I'll assume they'll stay green and land it.

**Pull request merged**